### PR TITLE
Add note about interpolated string overloads

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to dotnet/docs#26429.

*[Hide whitespace changes in diff.](https://github.com/dotnet/dotnet-api-docs/pull/7295/files?diff=split&w=1)*

### CODE REVIEW

File: xml/System.Diagnostics/Debug.xml [hunk 1]
```diff
@@ -51,43 +51,42 @@
...
-  
+      <format type="text/markdown"><![CDATA[
+
+## Remarks
+ If you use methods in the <xref:System.Diagnostics.Debug> class to print debugging information and check your logic with assertions, you can make your code more robust without impacting the performance and code size of your shipping product.
+
+ This class provides methods to display an <xref:System.Diagnostics.Debug.Assert%2A> dialog box, and to emit an assertion that will always fail. This class provides write methods in the following variations: <xref:System.Diagnostics.Debug.Write%2A>, <xref:System.Diagnostics.Debug.WriteLine%2A>, <xref:System.Diagnostics.Debug.WriteIf%2A>, and <xref:System.Diagnostics.Debug.WriteLineIf%2A>. .NET 6 and later versions have additional overloads.
```
Comment (tdykstra) [inline xml/System.Diagnostics/Debug.xml:59]: These aren't overloads, they're different methods, so "additional overloads" doesn't seem to fit.

File: xml/System.Diagnostics/Debug.xml [hunk 2]
```diff
@@ -272,27 +270,30 @@
...
-  
- When the application runs in user interface mode, it displays a message box that shows the call stack with file and line numbers. The message box contains three buttons: **Abort**, **Retry**, and **Ignore**. Clicking the **Abort** button terminates the application. Clicking **Retry** sends you to the code in the debugger if your application is running in a debugger, or offers to open a debugger if it is not. Clicking **Ignore** continues with the next instruction in the code.  
-  
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+This overload was introduced in .NET 6 to improve performance. In comparison to the overloads that take a `String` parameter, this overload only evaluates any interpolated string formatting items if the message is required.
```
Comment (tdykstra) [inline xml/System.Diagnostics/Debug.xml:277]: Should we prefer the language keyword, `string` over `String`?

**Review by tdykstra — APPROVED** [review]
LGTM

File: xml/System.Diagnostics/Debug.xml (hunk 2 above)
Comment (gewarren) [reply xml/System.Diagnostics/Debug.xml:277]: I don't think so, since that's C# specific.

File: xml/System.Diagnostics/Debug.xml (hunk 1 above)
Comment (gewarren) [inline xml/System.Diagnostics/Debug.xml:59]:
```suggestion
 This class provides methods to display an <xref:System.Diagnostics.Debug.Assert%2A> dialog box, and to emit an assertion that will always fail. This class provides write methods in the following variations: <xref:System.Diagnostics.Debug.Write%2A>, <xref:System.Diagnostics.Debug.WriteLine%2A>, <xref:System.Diagnostics.Debug.WriteIf%2A>, and <xref:System.Diagnostics.Debug.WriteLineIf%2A>.
```

Comment (gewarren) [reply xml/System.Diagnostics/Debug.xml]: I was referring to overloads of the listed methods, but upon rereading it, I don't think the addition added any useful information, so I've just removed it.

**Comment by gewarren** [comment]
Thanks for reviewing @tdykstra

**Pull request merged**